File: py/Euler104.py
from lib import isPanDigital

a = 1
b = 1
k = 2
while True:
    a += b
    k += 1
    h = str(a)[:9]
    t = str(a)[-9:]
    #print(h, t, len(str(a)))
    #print(isPanDigital(h), isPanDigital(t))
    if isPanDigital(h) and isPanDigital(t):
        print(a, k)
        break
    a, b = b, a

File: c/Euler104.c
#include <stdio.h>
#include <gmp.h>

int isPanDigital(char *s) {
    char d[10] = {0};
    int i;
    for (i = 0; i < 9; i++) {
        d[s[i]-'0']++;
    }
    for (i = 1; i <= 9; i++) {
        if (d[i] != 1) {
            return 0;
        }
    }
    return 1;
}

int main() {
    return 0;
}